packages/legend-query/src/components/QueryBuilderFilterPanel.tsx

```diff
@@ -261,7 +262,11 @@ const QueryBuilderFilterConditionEditor = observer(
                 </div>
               )}
               <QueryBuilderValueSpecificationEditor
-                valueSpecification={node.condition.value}
+                valueSpecification={observe_ValueSpecification(
```

Comment:
I think this works but ideally the way I see it is.
For value specification coming from the orginal `RawLambda`, those should be captured in the  observable after building the query state i.e QueryBuilderState `initialize` function. 
For value specifications build in the editor we should trace when we create new observable elements and wrap them with observables upon creation. 
For example here, we should wrap the observable when the FilterConditionState is created as this is where we create the new `AbstractPropertyExpression` in `buildPropertyExpressionFromExplorerTreeNodeData`